### PR TITLE
Add initial documentation for "new" Daml Shell

### PR DESCRIPTION
**Pull request by iggy-da**
Adds documentation for the "new" PQS-based Daml Shell under the Early Access menu section.

Since this is an interactive tool that heavily relies on colored output, with an interface that is likely to change from time to time, it contains screencast `gifs` that can be regenerated in an automated fashion from the Daml Shell repo.

### CODE REVIEW

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+setup instructions.
+
+Running Daml Shell
+------------------
+
+Daml Shell can be run from a ``jar`` artifact or from a ``Docker`` image
+(`tag
+list <https://hub.docker.com/repository/docker/digitalasset/daml-shell/tags?ordering=last_updated>`__).
+
+.. code:: sh
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:25]:
```suggestion
.. code-block:: sh
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+
+   docker run -it --net host docker.io/digitalasset/daml-shell:<version-tag> # change --net host if desired
+
+Commands
+--------
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:30]:
```suggestion
========
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+
+The easiest way to discover commands is to type the ``help`` command
+while Daml Shell is running.
+
+You can recurse into commands by typing
+``help <command> <sub-command> ...``. Auto-completion at the bottom of
+the screen will suggest possible command options.
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:37]:
```suggestion
``help <command> <sub-command> [...]``. Auto-completion at the bottom of
the screen suggests possible command options.
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+         * disconnect - close database connection
+         * exercise - choice by event-id
+         * exercises - exercised choices
+         * go - upper offset bound
+         * net-changes - net contract counts
+         * set - configure application settings
+         * transaction - transaction details
+
+FAQ
+===
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:70]:
```suggestion
***

.. _no-archived-contracts:
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+   **Why do I not see any archived contracts?**
+
+This is likely because PQS has been configured to seed the database from
+the ACS, which does not include historical offsets, archived contracts,
+or exercised choices.
+
+To see pre-existing archived contracts, the database needs to be seeded
+from the ``Transaction Stream`` or ``Transaction Tree Stream``.
+
+Solution: Set ``--pipeline-ledger-start`` to ``Genesis`` when running
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:81]:
```suggestion
If you don't see any archived contracts, PQS might be configured to seed the database from
the ACS, which does not include historical offsets, archived contracts,
or exercised choices.

To see pre-existing archived contracts, seed the database 
from the ``Transaction Stream`` or ``Transaction Tree Stream``.

For best results, set ``--pipeline-ledger-start`` to ``Genesis`` when running
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+PQS for the first time (refer to the `PQS
+documentation <https://docs.daml.com/query/pqs-user-guide.html>`__ or
+``--help`` output).
+
+   **Why do I not see any choices?**
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:86]:
```suggestion
Why don't I see any choices?
============================
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+
+Choices are only visible on the Ledger API’s
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:88]:
```suggestion
Choices are only visible on the Ledger API's
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 8]
```diff
@@ -0,0 +1,454 @@
...
+``Transaction Tree Stream``. Set ``--pipeline-datasource`` to
+``TransactionTreeStream`` when running PQS (refer to the `PQS
+documentation <https://docs.daml.com/query/pqs-user-guide.html>`__ or
+``--help`` output).
+
+   **Why do I still not see any choices?**
+
+See ``Why do I not see any archived contracts`` above.
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:96]:
```suggestion
If you still don't see choices, see :ref:`no-archived-contracts`
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 9]
```diff
@@ -0,0 +1,454 @@
...
+documentation <https://docs.daml.com/query/pqs-user-guide.html>`__ or
+``--help`` output).
+
+   **Why do all contracts show the same ledger offset?**
+
+See ``Why do I not see any archived contracts`` above.# Configuration
+
+Daml Shell can be configured via CLI arguments, environment variables,
+through a configuration file, or interactively using the ``set``
+command.
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:114]:
```suggestion
You can configure Daml Shell via CLI arguments, environment variables,
a configuration file, or interactively using the ``set``
command.
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 10]
```diff
@@ -0,0 +1,454 @@
...
+
+Command line arguments
+----------------------
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:117]:
```suggestion
======================
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 11]
```diff
@@ -0,0 +1,454 @@
...
+
+To see available command line options, run Daml Shell with the
+``--help`` flag:
+
+.. code:: sh
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:122]:
```suggestion
.. code-block:: sh
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 12]
```diff
@@ -0,0 +1,454 @@
...
+Configuration file
+------------------
+
+These configuration parameters can also be set in a reusable
+configuration file in HOCON (a JSON superset) using the flag
+``--config <filename>``.
+
+Example file content:
+
+.. code:: hocon
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:145]:
```suggestion
Configuration file
==================

You can set Daml Shell parameters in a reusable
configuration file in HOCON (a JSON superset) using the flag
``--config <filename>``.

Example file content:

.. code-block:: text
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 13]
```diff
@@ -0,0 +1,454 @@
...
+   identifier-hash-length = 20
+   wildcard-char = "…"
+   identifier-trim-location = "trailing"
+   full-identifiers = false
+   disable-color = false
+
+Interactively
+-------------
+
+Settings can be configured interactively using the ``set`` command.
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:159]:
```suggestion
Interactive configuration
=========================

You can configure settings interactively using the ``set`` command.
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 14]
```diff
@@ -0,0 +1,454 @@
...
+Example:
+
+::
+
+   > set identifier-hash-length full 
+   Disabled identifier shortening
+
+Type ``help set`` or ``help set <setting>`` to find out more about
+specific settings.# Usage
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:168]:
```suggestion
Type ``help set`` or ``help set <setting>`` to learn more about
specific settings.
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 15]
```diff
@@ -0,0 +1,454 @@
...
+Setup
+~~~~~
+
+Please refer to the `Getting Started <001-getting-started.md>`__ guide.
+
+Connecting
+~~~~~~~~~~
+
+We connect by entering the JDBC URL of our PQS postgres database. For
+example:
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:181]:
```suggestion

Usage
*****

This section provides some basic usage examples.

Connecting
==========

To connect, enter the JDBC URL of your PQS PostgreSQL database. For
example:
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 16]
```diff
@@ -0,0 +1,454 @@
...
+
+::
+
+   > connect jdbc:postgresql://localhost/postgres?user=postgres&password=postgres
+
+The status bar shows the connected status, the session offset range, and
+the datastore offset range.
+
+.. figure:: images/003-connect.gif?raw=true
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:190]:
```suggestion
.. figure:: images/003-connect.gif
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 17]
```diff
@@ -0,0 +1,454 @@
...
+``set identifier-hash-length full``, or set a custom length limit for
+hashes by running, for example, ``set identifier-hash-length 15``.
+
+Depending on the ledger implementation, offsets may be in hexadecimal
+format.
+
+The available offset range from the data-store is given in the
+``Datastore range`` status field. The offset range for Daml Shell to use
+to display payload counts and summaries is given by the
+``Session range`` status field.
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:207]:
```suggestion
=======

By default, offsets have leading zeroes removed. To see all identifiers
in full, including contract ID hashes, run
``set identifier-hash-length full`` or set a custom length limit for
hashes by running, for example, ``set identifier-hash-length 15``.

Depending on the ledger implementation, offsets may be in hexadecimal
format.

The available offset range from the datastore is specified in the
``Datastore range`` status field. The offset range for Daml Shell to use
to display payload counts and summaries is specified in the
``Session range`` status field.
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 18]
```diff
@@ -0,0 +1,454 @@
...
+
+::
+
+   > net-changes 
+   +3 PingPong:Cash [89a08f0324025f1…]
+   -3 PingPong:CashTransferProposal [89a08f0324025f1…]
+   +3 PingPong:IAsset [89a08f0324025f1…]
+   -3 PingPong:IAssetTransferProposal [89a08f0324025f1…]
+
+.. figure:: images/003-offset-commands.gif?raw=true
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:228]:
```suggestion
.. figure:: images/003-offset-commands.gif
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 19]
```diff
@@ -0,0 +1,454 @@
...
+   :alt: Connect
+
+Summary information
+~~~~~~~~~~~~~~~~~~~
+
+Commands such as ``active``, ``archives``, ``creates``, and
+``exercises`` can be used without argument to see payload counts by
+fully qualified identifier names. You can run ``help <command>`` to find
+out what each of them does.
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:237]:
```suggestion
fully qualified identifier names. For details, run ``help <command>``.
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 20]
```diff
@@ -0,0 +1,454 @@
...
+   +3 PingPong:Cash [89a08f0324025f1…]
+   -3 PingPong:CashTransferProposal [89a08f0324025f1…]
+   +3 PingPong:IAsset [89a08f0324025f1…]
+   -3 PingPong:IAssetTransferProposal [89a08f0324025f1…]
+
+.. figure:: images/003-offset-commands.gif?raw=true
+   :alt: Connect
+
+Summary information
+~~~~~~~~~~~~~~~~~~~
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:232]:
```suggestion
Summary information
===================
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 21]
```diff
@@ -0,0 +1,454 @@
...
+   │ PingPong:Cash [89a08f0324025f1…]        │ Template  │     6 │
+   ├─────────────────────────────────────────┼───────────┼───────┤
+   │ PingPong:IAsset [89a08f0324025f1…]      │ Interface │     6 │
+   ├─────────────────────────────────────────┼───────────┼───────┤
+   │ PingPong:IBounceable [89a08f0324025f1…] │ Interface │     2 │
+   ├─────────────────────────────────────────┼───────────┼───────┤
+   │ PingPong:Ping [89a08f0324025f1…]        │ Template  │     2 │
+   └─────────────────────────────────────────┴───────────┴───────┘
+
+.. figure:: images/003-summary-commands.gif?raw=true
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:254]:
```suggestion
.. figure:: images/003-summary-commands.gif

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 22]
```diff
@@ -0,0 +1,454 @@
...
+
+The package ID can be included in the FQN, in which case only payloads
+from that particular package will be returned, e.g.
+
+::
+
+   > active 89a08f0324025f1254f09edc0195ca24459c6302e88d2b9f636d2be5a615d1f1:PingPong:Ping
+
+or the package ID can be omitted, in which case payloads from all
+package IDs are returned, as long as they have the same name.
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:273]:
```suggestion
Payloads by fully qualified name
================================

Specify a fully qualified name (FQN) with the command ``active``,
``archives``, ``creates``, or ``exercises`` to list all
applicable payloads for that FQN.

To return payloads from a particular package only, include the package ID in
the FQN:

::

   > active 89a08f0324025f1254f09edc0195ca24459c6302e88d2b9f636d2be5a615d1f1:PingPong:Ping

If you omit the package ID, payloads from all
package IDs are returned, as long as they have the same name.
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 23]
```diff
@@ -0,0 +1,454 @@
...
+   ╞════════════╪══════════════════╪══════════════╪════════════════════════════════════════════════════════════════════════════════════╡
+   │ 8          │ 0022e89289bda36… │              │ label: one                                                                         │
+   │            │                  │              │ owner: Alice::12209038d324bf70625c580267d5957cb4c4c03bb7bce294713b48151a4a088afd3b │
+   ├────────────┼──────────────────┼──────────────┼────────────────────────────────────────────────────────────────────────────────────┤
+   │ a          │ 0093dce322a08c8… │              │ label: one copy updated                                                            │
+   │            │                  │              │ owner: Alice::12209038d324bf70625c580267d5957cb4c4c03bb7bce294713b48151a4a088afd3b │
+   └────────────┴──────────────────┴──────────────┴────────────────────────────────────────────────────────────────────────────────────┘
+
+The auto-completion provides both variants, FQN with and without package
+IDs.
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:289]:
```suggestion
The auto-completion provides both FQN variants (with and without package
ID).
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 24]
```diff
@@ -0,0 +1,454 @@
...
+
+Contract lookup
+~~~~~~~~~~~~~~~
+
+Contracts can be looked up by contract ID. Interface views will also be
+displayed, if any.
+
+The contract ID can be copied with the wildcard character (here “…”)
+included. The wildcard character will be expanded to any matching ID.
+
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:299]:
```suggestion
Contract lookup
===============

You can look up contracts by contract ID. Interface views are also
displayed, if any.

The contract ID can be copied with the wildcard character (here "...")
included. The wildcard character will be expanded to any matching ID.

```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 25]
```diff
@@ -0,0 +1,454 @@
...
+   ╟──────────────╫─────────────────────────────────────────╢
+   ║ Contract Key ║                                         ║
+   ╟──────────────╫─────────────────────────────────────────╢
+   ║ Payload      ║ ilabel: view one copy updated           ║
+   ╙──────────────╨─────────────────────────────────────────╜
+
+You can also compare two contracts in a ``diff``-style output format
+using the ``compare-contracts <id1> <id2>`` command.
+
+.. figure:: images/003-compare-contracts.gif?raw=true
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:343]:
```suggestion
.. figure:: images/003-compare-contracts.gif
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 26]
```diff
@@ -0,0 +1,454 @@
...
+The ``transaction`` command will show which contracts were created,
+which ones were archived, and what choices were exercised. It will also
+display the event ID for each of those events, as well as contract IDs
+and package IDs!
+
+.. figure:: images/003-transactions.gif?raw=true
+   :alt: compare-contracts
+
+Exercise lookup
+~~~~~~~~~~~~~~~
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:366]:
```suggestion
===============
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 27]
```diff
@@ -0,0 +1,454 @@
...
+
+Exercised choices can be looked up in the same manner as contracts,
+except that exercises are looked up by their event ID rather than by a
+contract ID. The commands for summaries and lookups mirror the
+functionality available for contracts.
+
+For example, exercise counts by FQN can be looked up like this:
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:373]:
```suggestion
For example, you can look up exercise counts by FQN:
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 28]
```diff
@@ -0,0 +1,454 @@
...
+   │ Identifier                                                    │ Type          │ Count │
+   ╞═══════════════════════════════════════════════════════════════╪═══════════════╪═══════╡
+   │ PingPong:AcceptIAssetTransferProposal [89a08f0324025f1254f0…] │ Consuming     │     6 │
+   ├───────────────────────────────────────────────────────────────┼───────────────┼───────┤
+   │ PingPong:ChangeLabel [89a08f0324025f1254f0…]                  │ Consuming     │     1 │
+   ├───────────────────────────────────────────────────────────────┼───────────────┼───────┤
+   │ PingPong:Copy [89a08f0324025f1254f0…]                         │ Non-consuming │     1 │
+   └───────────────────────────────────────────────────────────────┴───────────────┴───────┘
+
+Exercises for a specific choice can be looked up like this:
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:388]:
```suggestion
You can look up exercises for a specific choice:
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 29]
```diff
@@ -0,0 +1,454 @@
...
+   │ 12     │ 00bc1500e6c781559ad1… │          │ 00a8753f0ccbc503ab393dae99d7ca982b15f1429b61739c03dad299d123f21e5aca02122090c1f0e10da810f4ee1a961a91701863dd22f9f360f676725540929069cee17e │
+   ├────────┼───────────────────────┼──────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
+   │ 12     │ 001495835b90270f9699… │          │ 00ea78410739637ad2460e65ddf9382d2eef8d9a33a2ce38f912eaba065a342ebbca021220f5ffe34193c016a242a41735bf9670cf585a61e7423766bedbb15e92c6ea83be │
+   ├────────┼───────────────────────┼──────────┼────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
+   │ 12     │ 00844b2f4a2fb8ff73fe… │          │ 00bc875c5ee7cc6adb11a48a2cb4272e4374affa94f23cb8a7b358946f3bd5c4b4ca0212209cb5521de33a5a2d931e0b97a312753d772e55529bc086e0fb75376123131b6f │
+   └────────┴───────────────────────┴──────────┴────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+
+Individual exercises can be looked up by event ID:
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:409]:
```suggestion
To look up individual exercises, use the event ID:
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 30]
```diff
@@ -0,0 +1,454 @@
...
+   ╟──────────────╫────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╢
+   ║ Parent       ║                                                                                                                                            ║
+   ╟──────────────╫────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╢
+   ║ Argument     ║                                                                                                                                            ║
+   ╟──────────────╫────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╢
+   ║ Result       ║ 00a8753f0ccbc503ab393dae99d7ca982b15f1429b61739c03dad299d123f21e5aca02122090c1f0e10da810f4ee1a961a91701863dd22f9f360f676725540929069cee17e ║
+   ╙──────────────╨────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╜
+
+Setting offset bounds
+~~~~~~~~~~~~~~~~~~~~~
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:437]:
```suggestion
Setting offset bounds
=====================
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 31]
```diff
@@ -0,0 +1,454 @@
...
+
+The output of ``creates [<fqn>]`` and ``archives [<fqn>]`` can be
+bounded by ``set oldest`` (for the lower bound) and ``set latest`` (for
+the upper bound). ``go`` is an alias for ``set latest``.
+
+.. figure:: images/003-bounded-lookup.gif?raw=true
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:443]:
```suggestion
.. figure:: images/003-bounded-lookup.gif
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 32]
```diff
@@ -0,0 +1,454 @@
...
+   :alt: bounded lookup
+
+Finding transactions that created/archived a contract
+=====================================================
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:447]:
```suggestion
Finding transactions that created or archived a contract
========================================================
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 33]
```diff
@@ -0,0 +1,454 @@
...
+
+Once you know the offsets that a contract was created at, for example by
+using the ``archives`` command, you can look up the relevant
+transactions using the ``transaction at <offset>`` command.
+
+.. figure:: images/003-from-contract-to-transactions.gif?raw=true
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:453]:
```suggestion
Once you know the offsets that a contract was created at (for example, by
using the ``archives`` command), you can look up the relevant
transactions using the ``transaction at <offset>`` command.

.. figure:: images/003-from-contract-to-transactions.gif
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 34]
```diff
@@ -0,0 +1,454 @@
...
+.. SPDX-License-Identifier: Apache-2.0
+
+Daml Shell User Guide
+#####################
+
+Getting Started
+===============
+
+Prerequisites
+-------------
```
[the remark this answers is not in the document]
Comment (marciw-da) [reply docs/3.0.0/docs/tools/daml-shell/index.rst:11]: We use these heading symbols:
https://documentation-style-guide-sphinx.readthedocs.io/en/latest/style-guide.html#headings

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 35]
```diff
@@ -0,0 +1,454 @@
...
+documentation <https://docs.daml.com/query/pqs-user-guide.html>`__ or
+``--help`` output).
+
+   **Why do I still not see any choices?**
+
+See ``Why do I not see any archived contracts`` above.
+
+   **Why do I not see any interface views?**
+
+Interfaces are only visible on the Ledger API’s ``Transaction Stream``
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:100]:
```suggestion
Interfaces are only visible on the Ledger API's ``Transaction Stream``
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 36]
```diff
@@ -0,0 +1,454 @@
...
+``Transaction Tree Stream``. Set ``--pipeline-datasource`` to
+``TransactionTreeStream`` when running PQS (refer to the `PQS
+documentation <https://docs.daml.com/query/pqs-user-guide.html>`__ or
+``--help`` output).
+
+   **Why do I still not see any choices?**
+
+See ``Why do I not see any archived contracts`` above.
+
+   **Why do I not see any interface views?**
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:98]:
```suggestion
Why don't I see any interface views?
====================================
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 37]
```diff
@@ -0,0 +1,454 @@
...
+
+Interfaces are only visible on the Ledger API’s ``Transaction Stream``
+or ACS (not the ``Transaction Tree Stream``).
+
+Set ``--pipeline-datasource`` to ``TransactionStream`` when running PQS
+(refer to the `PQS
+documentation <https://docs.daml.com/query/pqs-user-guide.html>`__ or
+``--help`` output).
+
+   **Why do all contracts show the same ledger offset?**
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:108]:
```suggestion
Why do all contracts show the same ledger offset?
=================================================
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst (hunk 12 above)
Comment (marciw-da) [reply docs/3.0.0/docs/tools/daml-shell/index.rst:145]: I chose `text` for now, but you can use any of the Pygments lexers: https://pygments.org/docs/lexers/

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 38]
```diff
@@ -0,0 +1,454 @@
...
+   ╙──────────────╨─────────────────────────────────────────╜
+
+You can also compare two contracts in a ``diff``-style output format
+using the ``compare-contracts <id1> <id2>`` command.
+
+.. figure:: images/003-compare-contracts.gif?raw=true
+   :alt: compare-contracts
+
+Transaction lookup
+~~~~~~~~~~~~~~~~~~
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:347]:
```suggestion
Transaction lookup
==================
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 39]
```diff
@@ -0,0 +1,454 @@
...
+
+Transactions can be looked up either by transaction ID or by offset,
+i.e., by running ``transaction <transaction-id>`` or
+``transaction at <offset>`` respectively. Note the “``at``” when looking
+up by offset.
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:352]:
```suggestion
You can look up transactions by either transaction ID or offset,
by running ``transaction <transaction-id>`` or
``transaction at <offset>``, respectively. Note the ``at`` syntax when looking
up by offset.
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 40]
```diff
@@ -0,0 +1,454 @@
...
+
+The current transaction, at the head of the session offset range, can be
+displayed by just running ``transaction``.
+
+The ``transaction`` command will show which contracts were created,
+which ones were archived, and what choices were exercised. It will also
+display the event ID for each of those events, as well as contract IDs
+and package IDs!
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:360]:
```suggestion
To display the current transaction at the head of the session offset range,
run ``transaction``.

The ``transaction`` command shows which contracts were created,
which were archived, and what choices were exercised. It also
displays the event ID for each of those events, as well as contract IDs
and package IDs.
```

File: docs/3.0.0/docs/tools/daml-shell/index.rst [hunk 41]
```diff
@@ -0,0 +1,454 @@
...
+
+.. figure:: images/003-transactions.gif?raw=true
```
Comment (marciw-da) [inline docs/3.0.0/docs/tools/daml-shell/index.rst:362]:
```suggestion
.. figure:: images/003-transactions.gif
```

**Review by marciw-da — COMMENTED** [review]
I fixed some build issues and made changes for clarity, consistency, and Daml docs/Sphinx style.

****Important note!!**** This PR needs to move to the docs/draft folder rather than 3.0.0. The 3.0.0 docs are closed to new docs during code fork coordination. I will DM you w/ details, but the tl;dr is that if we put this in 3.0.0, we'll lose it. (Also as of 3.0.0, "early access" sections will move to the draft docs, so this move would happen anyway)

**Comment by iggy-da** [comment]
I've ported all the PR feedback upstream, with the exception of the introduction. I wrote conversion scripts (upstream) that address formatting issues, insert cross-references, etc.

The difference between the verbatim PR feedback and the generated version can be seen in the following commit: https://github.com/digital-asset/docs.daml.com/pull/621/commits/64c758a884e0a70872e11a180c30e2365a40f68c

----

There is one notable difference, which is that all
```rst
.. figure:: images/name.gif
   :alt: alt-text
```

image embeddings have been replaced with the following syntax

```rst
|name.gif|
  ...
.. |image.gif| image:: images/image.gif
```

for conversion reasons. Is there any issue with doing this?

**Comment by marciw-da** [comment]
> 
> There is one notable difference, which is that all
> 
> ```
> .. figure:: images/name.gif
>    :alt: alt-text
> ```
> 
> image embeddings have been replaced with the following syntax
> 
> ```
> |name.gif|
>   ...
> .. |image.gif| image:: images/image.gif
> ```
> 
> for conversion reasons. Is there any issue with doing this?

This seems to work fine in my local builds and previews, so it should be ok 👍

File: docs/draft/docs/tools/daml-shell/index.rst [hunk 42]
```diff
@@ -0,0 +1,451 @@
...
+
+Finding transactions that created or archived a contract
+========================================================
+
+Once you know the offsets that a contract was created at (for example,
+by using the ``archives`` command), you can look up the relevant
+transactions using the ``transaction at <offset>`` command.
+
+|003-from-contract-to-transactions.gif| # FAQ
+
```
Comment (marciw-da) [inline docs/draft/docs/tools/daml-shell/index.rst:399]:
```suggestion
|003-from-contract-to-transactions.gif| 

FAQ
***
```

**Review by marciw-da — COMMENTED** [review]
Thanks for making all these changes! Looks good except the FAQ heading didn't get converted from .md to .rst.

File: docs/2.9.0/docs/tools/daml-shell/index.rst [hunk 43]
```diff
@@ -0,0 +1,454 @@
+.. Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
```
Comment (marciw-da) [inline docs/2.9.0/docs/tools/daml-shell/index.rst:3]:
```suggestion
.. SPDX-License-Identifier: Apache-2.0

.. enterprise-only::

```

Comment (marciw-da) [reply docs/2.9.0/docs/tools/daml-shell/index.rst:3]: Missed this in the other reviews -- doesn't Daml Shell require an enterprise license? 

This directive adds the enterprise license disclaimer, like the one on [this page](https://docs.daml.com/canton/usermanual/domains/ethereum.html) (and many others)

It looks like this in local preview:

Screenshot 2024-02-15 at 11 44 56 AM


👉 (but if I'm wrong about Daml Shell being an enterprise feature, reject this change!) 🙅

**Review by marciw-da — APPROVED** [review]
Approving to unblock you, but note that I proposed adding an enterprise license disclaimer (if that's correct)

Thanks again for working through all the Sphinx intricacies 🙂

File: docs/2.9.0/docs/tools/daml-shell/index.rst (hunk 43 above)
Comment (hrischuk-da) [reply docs/2.9.0/docs/tools/daml-shell/index.rst:3]: It does require an enterprise license.

**Comment by iggy-da** [comment]
Thanks, I added the header to the upstream rst generation

**Pull request closed without merging**